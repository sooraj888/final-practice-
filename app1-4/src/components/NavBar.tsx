import { AppBar, Toolbar, Typography } from "@material-ui/core";
import React from "react";

const NavBar = (): JSX.Element => {
  return (
    <AppBar position="static">
      <Toolbar>
        <Typography>Country Weather App</Typography>
      </Toolbar>
    </AppBar>
  );
};

export default NavBar;
