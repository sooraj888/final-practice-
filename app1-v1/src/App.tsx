import React from "react";
import logo from "./logo.svg";
import "./App.css";
import { Box } from "@material-ui/core";

function App() {
  return <Box className="App"></Box>;
}

export default App;
