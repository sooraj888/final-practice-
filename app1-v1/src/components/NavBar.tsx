import { AppBar, Toolbar, Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";
import React from "react";

const useStyles = makeStyles({
  roots: {
    fontSize: "40px",
  },
});

const NavBar = (): JSX.Element => {
  const classes = useStyles();
  return (
    <AppBar position="static">
      <Toolbar>
        <Typography className={classes.roots}>Country Weather App</Typography>
      </Toolbar>
    </AppBar>
  );
};

export default NavBar;
