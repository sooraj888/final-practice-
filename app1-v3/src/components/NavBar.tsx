import { AppBar, Toolbar, Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";
import React from "react";

const useStyle = makeStyles({
  root: {
    fontSize: "40px",
  },
});

const NavBar = (): JSX.Element => {
  const classes = useStyle();
  return (
    <AppBar position="static">
      <Toolbar>
        <Typography className={classes.root}>
          Country and Weather App
        </Typography>
      </Toolbar>
    </AppBar>
  );
};

export default NavBar;
