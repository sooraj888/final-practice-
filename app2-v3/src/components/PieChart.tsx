import { Box } from "@material-ui/core";
import React from "react";

type PieChartType = { percent: number };

const PieChart = ({ percent }: PieChartType): JSX.Element => {
  return (
    <div
      style={{
        display: "flex",
        width: "200px",
        height: "200px",
        background: `conic-gradient(green ${percent}% ,red 0%)`,
        borderRadius: "50%",
      }}
    ></div>
  );
};

export default PieChart;
