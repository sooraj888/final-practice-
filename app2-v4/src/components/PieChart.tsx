import { Box } from "@material-ui/core";
import React from "react";
type PieChartType = { precent: number };
const PieChart = ({ precent }: PieChartType): JSX.Element => {
  return (
    <div
      style={{
        display: "flex",
        height: "200px",
        width: "200px",
        background: `conic-gradient(green ${precent}%, red 0%) `,
        borderRadius: "50%",
      }}
    ></div>
  );
};

export default PieChart;
