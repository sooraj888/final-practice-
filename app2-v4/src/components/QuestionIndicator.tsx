import { Box } from "@material-ui/core";
import React from "react";
import { Link } from "react-router-dom";
import questionData from "./../questionData.json";

export type QuestionIndicatorType = {
  answerArray: Array<string | Array<string>>;
  id: number;
};
const QuestionIndicator = ({
  answerArray,
  id,
}: QuestionIndicatorType): JSX.Element => {
  return (
    <Box
      sx={{ display: "flex", justifyContent: "space-around", margin: "30px" }}
    >
      {questionData.map((item: any, key): JSX.Element => {
        const selected = item.id === id;
        const completed = answerArray?.[key].length > 0;
        return (
          <Circle
            key={key}
            id={item.id}
            completed={completed}
            selected={selected}
          ></Circle>
        );
      })}
    </Box>
  );
};

type QuestionIndicator = { id: number; selected: boolean; completed: boolean };
export default QuestionIndicator;

const Circle = ({
  id,
  selected,
  completed,
}: QuestionIndicator): JSX.Element => {
  return (
    <Link to={`/question/${id}`}>
      <Box
        sx={{
          display: "flex",
          bgcolor: completed ? "red" : "gray",
          width: "50px",
          height: "50px",
          borderRadius: "50%",
          justifyContent: "center",
          alignItems: "center",
          border: selected ? "5px solid blue" : "",
        }}
      >
        {id}
      </Box>
    </Link>
  );
};
