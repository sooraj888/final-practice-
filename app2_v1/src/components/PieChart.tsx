import { Box } from "@material-ui/core";
import React from "react";

const PieChart = ({ firstColorPercent }: any) => {
  return (
    <div
      style={{
        width: "200px",
        height: "200px",
        background: `conic-gradient(green ${firstColorPercent}%, red 0% )`,
        borderRadius: "50%",
      }}
    ></div>
  );
};

export default PieChart;
