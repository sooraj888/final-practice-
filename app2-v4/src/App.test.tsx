import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";
import NavBar from "./components/NavBar";
import PieChart from "./components/PieChart";
import QuestionIndicator from "./components/QuestionIndicator";
import { BrowserRouter } from "react-router-dom";

describe("test on NavBar", () => {
  test("rendering NavBar", () => {
    render(<NavBar />);
    screen.debug();
  });
});

describe("test on NavBar", () => {
  test("snapshot test NavBar", () => {
    const { container } = render(<NavBar />);
    expect(container).toMatchSnapshot();
  });
});

describe("test on NavBar", () => {
  test("finding element NavBar", () => {
    render(<NavBar />);
    expect(screen.getByText("Question and Weather App")).toBeInTheDocument();
  });
});

describe("test on NavBar", () => {
  test("finding element NavBar", () => {
    render(<NavBar />);
    expect(screen.getByText("Question and Weather App")).toHaveAttribute(
      "class"
    );
  });
});

describe("test on PieChart", () => {
  test("rendering PieChart", () => {
    render(<PieChart precent={50} />);
    screen.debug();
  });
});

describe("test on PieChart", () => {
  test("snapshot test PieChart", () => {
    const { container } = render(<PieChart precent={50} />);
    expect(container).toMatchSnapshot();
  });
});

describe("test on QuestionIndicator", () => {
  test("rendering QuestionIndicator", () => {
    render(
      <BrowserRouter>
        <QuestionIndicator answerArray={["2", "4", "", ""]} id={1} />
      </BrowserRouter>
    );
    screen.debug();
  });
});

describe("test on QuestionIndicator", () => {
  test("snapshot test QuestionIndicatorR", () => {
    const { container } = render(
      <BrowserRouter>
        <QuestionIndicator answerArray={["2", "4", "", ""]} id={1} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});

describe("test on NavBar", () => {
  test("finding element NavBar", () => {
    render(
      <BrowserRouter>
        <QuestionIndicator answerArray={["2", "4", "", ""]} id={1} />
      </BrowserRouter>
    );
    expect(screen.getByText("1")).toBeInTheDocument();
  });
});

describe("test on NavBar", () => {
  test("finding element NavBar", () => {
    render(
      <BrowserRouter>
        <QuestionIndicator answerArray={["2", "4", "", ""]} id={1} />
      </BrowserRouter>
    );
    expect(screen.getByText("2")).toHaveAttribute("class");
  });
});
