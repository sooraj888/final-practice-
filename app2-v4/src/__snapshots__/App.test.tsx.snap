// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on NavBar snapshot test NavBar 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Question and Weather App
      </p>
    </div>
  </header>
</div>
`;

exports[`test on PieChart snapshot test PieChart 1`] = `
<div>
  <div
    style="display: flex; height: 200px; width: 200px; border-radius: 50%;"
  />
</div>
`;

exports[`test on QuestionIndicator snapshot test QuestionIndicatorR 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-6"
  >
    <a
      href="/question/1"
    >
      <div
        class="MuiBox-root MuiBox-root-7"
      >
        1
      </div>
    </a>
    <a
      href="/question/2"
    >
      <div
        class="MuiBox-root MuiBox-root-8"
      >
        2
      </div>
    </a>
    <a
      href="/question/3"
    >
      <div
        class="MuiBox-root MuiBox-root-9"
      >
        3
      </div>
    </a>
    <a
      href="/question/4"
    >
      <div
        class="MuiBox-root MuiBox-root-10"
      >
        4
      </div>
    </a>
  </div>
</div>
`;
